api-admin/src/test/java/bio/terra/pearl/api/admin/service/ConfigExtServiceTests.java

```diff
@@ -28,8 +32,20 @@ public void testConfigMap() {
     when(b2CConfiguration.clientId()).thenReturn("client123");
     when(b2CConfiguration.policyName()).thenReturn("policy123");
     ConfigExtService configExtService =
-        new ConfigExtService(b2CConfiguration, applicationRoutingPaths);
+        new ConfigExtService(b2CConfiguration, applicationRoutingPaths, pepperDSMConfig);
     Map<String, String> configMap = configExtService.getConfigMap();
     Assertions.assertEquals("something.org", configMap.get("participantUiHostname"));
   }
+
+  @Test
+  public void testInternalConfigRequiresSuperuser() {
+    AdminUser user = AdminUser.builder().superuser(false).build();
+    ConfigExtService configExtService =
+        new ConfigExtService(b2CConfiguration, applicationRoutingPaths, pepperDSMConfig);
+    Assertions.assertThrows(
+        PermissionDeniedException.class,
+        () -> {
+          configExtService.getInternalConfigMap(user);
+        });
+  }
```

Comment:
Is it straightforward to provide `PepperDSMConfig` values to test the actual assembly? (Sorry, I have not used MockBean, so I do not know what that provides.)

Reply:
yeah, we could, I was running fast and loose with this because I want to get it working, and I also always forget how MockBeans work :)